// This file copyright (c) 2011-2015, Quickstep Technologies LLC.
// Copyright (c) 2015, Pivotal Software, Inc.
// All rights reserved.
// See file CREDITS.txt for details.

syntax = "proto2";

package quickstep.serialization;

import "types/Type.proto";

message TypedValue {
  required Type.TypeID type_id = 1;

  // NOTE(zuyu): For a NULL value, none of the optional fields are filled in.
  optional int32 int_value = 2;
  optional int64 long_value = 3;
  optional float float_value = 4;
  optional double double_value = 5;
  optional bytes out_of_line_data = 6;
  optional int64 datetime_value = 7;
  optional int64 datetime_interval_value = 8;
  optional int64 year_month_interval_value = 9;
}
